### PR TITLE
Support parsing `DROP PUBLICATION` in PostgreSQL

### PR DESCRIPTION
**Pull request by everly-gif**
Refers #14015 

Changes proposed in this pull request:
- Adds Support for parsing `DROP PUBLICATION` in `PostgreSQL`
- Adds Tests

✅ Builds locally

### CODE REVIEW

**Comment by everly-gif** [comment]
3 CI get cancelled without throwing any errors , I don't think it has to do with my PR.
- Continuous Integration / JDK 8. 
- Continuous Integration / JDK 11
- Continuous Integration / JDK 17
Would like someone's input.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/ddl/DropPublicationStatement.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.sql.parser.sql.common.statement.ddl;
+
+import lombok.ToString;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.AbstractSQLStatement;
+
+/**
+ * Drop  publication  statement.
```
Comment (strongduanmu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/ddl/DropPublicationStatement.java:24]: @everly-gif Please remove extra whitespace.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/ddl/DropPublicationStatement.java (hunk 1 above)
Comment (everly-gif) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/ddl/DropPublicationStatement.java]: Sure, removed

**Review by strongduanmu — APPROVED** [review]
@everly-gif Good job, merged.

**Pull request merged**